scrum-gang/neutralinojs
# chore: improve CMake builds on macOS and Linux

## Description
This PR aims to make CMake build the actual function app by globbing all files in `core-linux/**/*.cpp` and `core-macOS/**/*.cpp`.

## Changes proposed
- Globs additional files.
- Puts out fires to make everything compile. Note that I moved some stuff out of `core-shared` into `core-linux`, so this will probably break the Windows code. We'll have to copy/paste that manually for now.

## How to test it

On macOS/Linux:
```sh
$ cd build
$ cmake ..
$ make
```

The `neutralino` binary should be produced. Running it directly in the `build` folder should output an error about a missing `settings.json`. To provide a minimal environment for the binary, download the latest Neutralino release and drop the binary in that folder. It should run:

<img width="1112" alt="image" src="https://user-images.githubusercontent.com/16010076/69495976-0e3fbf80-0e9b-11ea-84a2-e17ff65212de.png">

On Linux Docker Container:
```sh
$ docker build -t . test
$ docker run test
```

So far, I haven't actually tested the output binary. However, linking all of the missing `.cpp` files seems like the right thing to do...

## Next steps
Do the same for Windows.

## Deploy notes


None.
